### ISSUE TITLE
Settings: first Dialog opens in light unreadable mode (white text on white background) when in dark mode 

### ISSUE THREAD

**Issue by eddiemuc** [cgeo/cgeo#10904]
The first Dialog opens in light unreadable mode (white text on white background) when in dark mode. The second dialog opens fine after that.

I discovered this while working in PR #10894 but had no real idea how to fix it, thus this issue.

Reproduce:
* set c:geo to dark mode
* restart c:geo
* Go into the "Settings"
* Open any dialog (e.g. "System" -> "Generate Logfile")

--> The dialog shows up unreadable (white text on white background)

* Close/cancel the dialog, open the same again or another one (e.g. again "System" -> "Generate Logfile")

--> The dialog shows up fine

First time:

![image](https://user-images.githubusercontent.com/6909759/121767737-8b548780-cb5a-11eb-9f8c-0d7b30e500db.png)

Second time:
![image](https://user-images.githubusercontent.com/6909759/121767742-95768600-cb5a-11eb-88a9-2ff38d0d819b.png)

**Comment by Lineflyer** [comment]
Same behavior reported by a user of our nightly builds for the following dialogs:

- Select map source from settings
- Create backup (overwrite warning)

![grafik](https://user-images.githubusercontent.com/949669/130460970-26f66687-f54a-4012-92cc-a813b2de574e.png)

![grafik](https://user-images.githubusercontent.com/949669/130461003-5b0b8f00-22b1-4b13-8bad-f22dd143ef3d.png)


Only once per session. Second access is OK.

System info:
```
## System information

c:geo version: 2021.08.19-NB-14e9ae1

Device:
-------
- Device type: SM-G950F (dreamltexx, samsung)
- Available processors: 8
- Android version: 9
- Android build: PPR1.180610.011.G950FXXUCDUD1
- Screen resolution: 1440x2792px (411x773dp)
- Sailfish OS detected: false
- Google Play services: enabled - 21.26.21 (100408-387928701)
- HW acceleration: enabled (default state)

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: orientation

Program settings:
-------
- Settings: v8, Count:140
- Set language: de_DE (system default)
- System date format: dd.MM.yy
- Time zone: GMT+02:00
- Debug mode active: no
- Last backup: 19. Aug., 08:51
- Routing mode: Walk
- Live map mode: true
- OSM multi-threading: true / threads: 4
- Map: Germany (Offline)
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMapSource:6364-3564:Offlinekarten/germany.map
  - Atts: Map data (c) OpenStreetMap contributors
  - Theme: none

Filters:
-------
- Hide waypoints: -
- LIVE: Cache-Typ:11 Elemente,Status:Gefunden=Nein ([:inconclusive=false:advanced=false]AND(type:CITO:MULTI:EARTH:WHERIGO:VIRTUAL:MYSTERY:EVENT:WEBCAM:USER_DEFINED:LETTERBOX:TRADITIONAL;difficulty_terrain:d=-:d=-:t=-:t=-;status:found_no))
- OFFLINE: Keine ([:inconclusive=false:advanced=false])

Services:
-------
- Geocaching sites enabled:
   geocaching.com: Logged in (Anmeldung OK) / PREMIUM
   Geocaching.com Adventure Lab
- Geocaching.com date format: dd.MM.yyyy
- Routing: external / BRouter installed: false
- Installed c:geo plugins: contacts

Permissions & paths:
-------
- Fine location permission: granted
- Write external storage permission: granted
- System internal c:geo dir: /data/user/0/cgeo.geocaching (6,2 GB free) v2 internal isDir(8 entries)
- Legacy User storage c:geo dir: /storage/emulated/0/cgeo (6,1 GB free) v2 external non-removable isDir(0 entries)
- Geocache data: /storage/6364-3564/Android/data/cgeo.geocaching/files/GeocacheData (76,6 GB free) v2 external removable isDir(2635 entries)
- Internal theme sync (is turned off): /data/user/0/cgeo.geocaching/MapThemeData (6,2 GB free) v2 internal isDir(0 entries)
- Public Folders: #11
  - BASE: SD-Karte/Android/data/cgeo.geocaching (User-Defined)[SD-Karte/Android/data/cgeo.geocaching[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching::]] (Uri: content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching/document/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching, Av:true, files:>=420, dirs:>=2644, totalFileSize:>=12,6 MB, free space: 76,6 GB, files on device: 0)
  - OFFLINE_MAPS: SD-Karte/Offlinekarten (User-Defined)[SD-Karte/Offlinekarten[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/6364-3564%3AOfflinekarten::]] (Uri: content://com.android.externalstorage.documents/tree/6364-3564%3AOfflinekarten/document/6364-3564%3AOfflinekarten, Av:true, files:16, dirs:2, totalFileSize:8,5 GB, free space: 76,6 GB, files on device: 0)
  - OFFLINE_MAP_THEMES: SD-Karte/Offlinekarten/_themes (Default)[SD-Karte/Offlinekarten/_themes[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/6364-3564%3AOfflinekarten::/_themes]] (Uri: content://com.android.externalstorage.documents/tree/6364-3564%3AOfflinekarten/document/6364-3564%3AOfflinekarten%2F_themes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 76,6 GB, files on device: 0)
  - LOGFILES: SD-Karte/Android/data/cgeo.geocaching/logfiles (Default)[SD-Karte/Android/data/cgeo.geocaching/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching/document/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching%2Flogfiles, Av:true, files:2, dirs:0, totalFileSize:345,6 KB, free space: 76,6 GB, files on device: 0)
  - GPX: SD-Karte/Android/data/cgeo.geocaching/gpx (Default)[SD-Karte/Android/data/cgeo.geocaching/gpx[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching::/gpx]] (Uri: content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching/document/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching%2Fgpx, Av:true, files:349, dirs:0, totalFileSize:46,5 MB, free space: 76,6 GB, files on device: 0)
  - BACKUP: SD-Karte/Android/data/cgeo.geocaching/backup (Default)[SD-Karte/Android/data/cgeo.geocaching/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching::/backup]] (Uri: content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching/document/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching%2Fbackup, Av:true, files:8, dirs:4, totalFileSize:2,6 GB, free space: 76,6 GB, files on device: 0)
  - FIELD_NOTES: SD-Karte/Android/data/cgeo.geocaching/field-notes (Default)[SD-Karte/Android/data/cgeo.geocaching/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching/document/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching%2Ffield-notes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 76,6 GB, files on device: 0)
  - SPOILER_IMAGES: SD-Karte/Android/data/cgeo.geocaching/GeocachePhotos (Default)[SD-Karte/Android/data/cgeo.geocaching/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching/document/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching%2FGeocachePhotos, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 76,6 GB, files on device: 0)
  - ROUTING_BASE: SD-Karte/Android/data/cgeo.geocaching/routing (Default)[SD-Karte/Android/data/cgeo.geocaching/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching::/routing]] (Uri: content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching/document/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching%2Frouting, Av:true, files:7, dirs:1, totalFileSize:81,2 KB, free space: 76,6 GB, files on device: 0)
  - ROUTING_TILES: SD-Karte/Android/data/cgeo.geocaching/routing/segments4 (Default)[SD-Karte/Android/data/cgeo.geocaching/routing/segments4[PERSISTABLE_FOLDER(ROUTING_BASE)#1:p-content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching::/routing/segments4]] (Uri: content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching/document/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching%2Frouting%2Fsegments4, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 76,6 GB, files on device: 0)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 6,2 GB, files on device: -1)
- Map render theme path:
- PersistedDocumentUris: #1
- TRACK: null
- Persisted Uri Permissions: #2
  - content://com.android.externalstorage.documents/tree/6364-3564%3AOfflinekarten (21. Feb., 20:55):RW
  - content://com.android.externalstorage.documents/tree/6364-3564%3AAndroid%2Fdata%2Fcgeo.geocaching (3. März, 21:39):RW
- Database: /storage/emulated/0/Android/data/cgeo.geocaching/files/databases/data (v96, Size:673,7 MB) on user storage

--- End of system information ---
```

Ticket: 242261

**Comment by moving-bits** [comment]
Haven't encountered this any more since migration to AndroidX preferences.
Feel free to reopen this or create a new issue if it or something similar happens again.
